cmd/serverless-init/main.go

```diff
@@ -79,6 +80,12 @@ func setupMetricAgent(tags map[string]string) *metrics.ServerlessMetricAgent {
 	return metricAgent
 }
 
+func flushMetricsAgent(metricAgent *metrics.ServerlessMetricAgent) {
+	for range time.Tick(3 * time.Second) {
```

Comment:
Do we want this to be configurable by env var or config file?

Reply:
Oh also is there a reason we use this and not a [Ticker](https://gobyexample.com/tickers)?

Reply:
I think this method uses a Ticker under the hood, (time.Tick produces a channel). The for range syntax applied to a channel works the same as a select statement. So it's roughly equivalent.

Reply:
Good questions! I propose we leave it 3 seconds to match the current Trace Flusher implementation. I think we can revisit this in the future but we're just looking to get something out to fix this issue for Cloud Run and Container App users.

For the second one, my understanding is that Ticker runs every 3 seconds on the dot. For example, if flushing takes 1 second, it'll wait 2s (instead of sleeping for 3). I don't think this behavior matters too much -- we're not trying to wait **exactly** 3 seconds, we just need some type of buffer time to flush. imo there's also less cognitive overhead and the function is relatively simple so I think a goroutine with a sleep is an okay solution here.